tool/tsh/app_cloud.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+// cloudApp is a dummy interface shared between awsApp, azureApp and gcpApp
+type cloudApp interface {
+	dummyCloudAppMethod()
+}
+
+// cloudAppImpl automatically implements cloudApp when inherited
+type cloudAppImpl struct {
+}
+
+func (_ cloudAppImpl) dummyCloudAppMethod() {
+}
```

Comment:
Why do we need this dummy interface ? 


Can we just extract the CloudApp common  interfaces for the aws/gcp/azure cloud apps like: 
```go
type CloudAppHandler interface {
  StartLocalProxies() error
  Close() error
  GetCLIConfig() CLIConf
  RunCommand() error
  GetEnvVars() map[string]string, error
 
 Match(app tlsca.RouteToApp) 
 GetFriendlyName() string
}
```
to get more readability ?

Reply:
Refactored to use generics.